### PR TITLE
hip-0000-free-network-info-query.md

### PR DESCRIPTION
**Pull request by SimiHunjan**

### CODE REVIEW

File: HIP/hip-0000-free-network-info-query.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+- IP address
+- Port number
+- Node account ID
+- Node ID
+- Node certificate hash
+- RSA_PublicKey
+- Description
+- Stake
+
+Today, applications have to use the File Service and submit two file contents query requests to return all the required node information. This results in two transaction fees for processing each of the queries. The information contained within the address book files are required for clients to send Hedera transactions or queries to valid Hedera consensus nodes.
```
Comment (sergmetelin) [inline HIP/hip-0000-free-network-info-query.md:22]: `The information contained within the address book files **_is_** required`

File: HIP/hip-0000-free-network-info-query.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+## Rationale 
+
+The proposal is to create a free non-file HAPI query to return network information. As we continue to grow the network and transition into a permissionless model, the number of times the local address book will be required to update will increase. The transaction fee associated with current file query hinders the ability to update the address book as frequently as it might be needed for applications to ensure their address book is always up to date. This information should be free.
```
Comment (sergmetelin) [inline HIP/hip-0000-free-network-info-query.md:26]: `with **the** current`

File: HIP/hip-0000-free-network-info-query.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+| hip               | title              | author         | type            | status         | created       | discussions |
+| ----------- | -----------  | ----------- | ----------- |----------- | ----------- | ----------- |
+| TBD            | Free network info query | Simi Hunjan (@SimiHunjan ) |Service| Draft | 2021-06-09   |   https://github.com/hashgraph/hedera-improvement-proposal/discussions/82
```
Comment (sergmetelin) [inline HIP/hip-0000-free-network-info-query.md:3]: Do you mind renaming the file to `hip-20.md` and adding `hip-20` in the header table?

File: HIP/hip-0000-free-network-info-query.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+- A client could spam the network since the query is free and disable other users from processing their request 
+
+## Open Issues
+
+- Throttle the query to prevent spamming of the network
+
+## How to Teach This
+
+- Comments in the protobuf files in the [hedera-protobufs](https://github.com/hashgraph/hedera-protobufs) repository describing the nature of query
```
Comment (sergmetelin) [inline HIP/hip-0000-free-network-info-query.md:55]: `the nature of **the** query`

**Review by sergmetelin — CHANGES_REQUESTED** [review]

File: HIP/hip-0000-free-network-info-query.md (hunk 1 above)
Comment (SimiHunjan) [reply HIP/hip-0000-free-network-info-query.md]: resolved

File: HIP/hip-0000-free-network-info-query.md (hunk 3 above)
Comment (SimiHunjan) [reply HIP/hip-0000-free-network-info-query.md]: updated to hip 21

File: HIP/hip-0000-free-network-info-query.md (hunk 4 above)
Comment (SimiHunjan) [reply HIP/hip-0000-free-network-info-query.md]: resolved

File: HIP/hip-0000-free-network-info-query.md (hunk 2 above)
Comment (SimiHunjan) [reply HIP/hip-0000-free-network-info-query.md]: resolved

**Review by sergmetelin — APPROVED** [review]

**Pull request merged**